frontend/hub/approvals/hooks/useApprovalActions.tsx

```diff
@@ -57,7 +57,8 @@ export function useApprovalActions(callback?: (collections: CollectionVersionSea
         onClick: (collection) => approveCollections([collection]),
         isDanger: false,
         isDisabled: (collection) =>
-          can_upload_signatures && require_upload_signatures && collection.is_signed
+          (can_upload_signatures && require_upload_signatures && collection.is_signed) ||
+          collection?.repository?.pulp_labels?.pipeline === 'approved'
             ? t`You do not have rights to this operation`
             : undefined,
```

Comment:
Looks like we may also want to separate the messages, not sure if you want to do it now...

Something like..

```
       (can_upload_signatures && require_upload_signatures && collection.is_signed) ? t`...rights...` :
       collection?.repository?.pulp_labels?.pipeline === 'approved' ? t`...already there...` : undefined
```

... except that `can_upload_signatures && require_upload_signatures && collection.is_signed` is also a bug we should fix...  Approval should absolutely be possible even if a collection is already signed.

It looks like it could be a misinterpretation of the insights mode check where a collection must be signed before it can be approved.


Sooo... the button should be disabled when ANY of these is true:

* pipeline is approved (`collection?.repository?.pulp_labels?.pipeline === 'approved'`)
* a signature is required but not present (`can_upload_signatures && require_upload_signatures && !collection.is_signed` (note the extra `!`))


What about

```js
  (collection?.repository?.pulp_labels?.pipeline === 'approved') ? true :
  (can_upload_signatures && require_upload_signatures && !collection.is_signed) ? t`Signature must be uploaded first` : undefined
```